plugins/modules/ec2_vol.py

```diff
@@ -356,22 +355,40 @@ def update_volume(module, ec2_conn, volume):
     req_obj = {'VolumeId': volume['volume_id']}
 
     if module.params.get('modify_volume'):
+        target_type = module.params.get('volume_type')
+        original_type = None
+        type_changed = False
+        if target_type:
+            original_type = volume['volume_type']
+            if target_type != original_type:
+                type_changed = True
+                req_obj['VolumeType'] = target_type
+
         iops_changed = False
-        if volume['volume_type'] != 'standard':
-            target_iops = module.params.get('iops')
-            if target_iops:
-                original_iops = volume['iops']
-                if target_iops != original_iops:
+        target_iops = module.params.get('iops')
```

Comment:
```suggestion
        target_iops = int(module.params.get('iops'))
```
What do you think about casting to int? I think this would avoid similar issues https://github.com/ansible-collections/community.aws/issues/570?

Reply:
The param explicitly defines itself as an int.  Casting shouldn't be required.

Part of the problem with  ansible-collections/community.aws#570 is that it's a sub-option and the parent was simply defined as a dict without defining the suboptions and setting the type.

Reply:
Oh, you're right! I retire my suggestions!